### PR TITLE
Restore sidebar in focus mode on Pattern click through in Browse Mode `Library` 

### LINKED ISSUE #51839: Library click through to template part focus mode hides sidebar and Navigation editing feature
As part of Navigation on Browse Mode it was agreed that in focus mode for Template Parts it would be useful to [display any Navigation Menus present in the blocks within in the browse mode sidebar](https://github.com/WordPress/gutenberg/pull/51492).

However with the change to Library, when you click on a Template Part you now get taken to the focus mode in "edit" mode (`canvas=edit`) which hides the sidebar.

Given the value of https://github.com/WordPress/gutenberg/pull/51492, we should retain the sidebar. If necessary this can be done conditionally by detecting whether a Navigation block is present in the blocks.

https://github.com/WordPress/gutenberg/assets/444434/3f0d2666-697b-40da-98f1-7c22564f3aaf

### PR DESCRIPTION
**Pull request by getdave**
## What?

Updates the link when clicking through from a Browse Mode Library item into focus mode to default to `view` mode rather than `edit`.

Fixes https://github.com/WordPress/gutenberg/issues/51839

## Why?

As detailed in the associated Issue, not showing the sidebar means https://github.com/WordPress/gutenberg/pull/51492 is effectively hidden from the user. 

This PR restores it to be a feature users will actually encounter.

## How?

Sets route to `view` rather than `edit`.

## Testing Instructions

- Make sure your `Header` template part / pattern has a Navigation block with a menu in it.
- Go to `LIbrary` in Browse mode and click through to your `Header` pattern.
- See the sidebar is open and the url has `canvas=view` in it.

### CODE REVIEW

**Review by carolinan — APPROVED** [review]
For what it's worth, I much prefer this flow.

**Comment by aaronrobertshaw** [comment]
> For what it's worth, I much prefer this flow.

My understanding is this wasn't the flow when the [design video](https://github.com/WordPress/gutenberg/issues/50028#issuecomment-1550777201) for the library was shared. That design was what we worked towards but it appears that before the library landed the navigation editing work changed the flow to always go via view mode. 

I don't have any strong feelings towards this. Each pattern in the library went to edit mode simply due to consistency with how it originally worked and the design. 

The navigation editing feature in the sidebar does present a valid reason to switch the patterns to go via view mode.

**Comment by getdave** [comment]
@aaronrobertshaw Yeh it seems like there was some inconsistency in the designs around Library and Navigation sections. All good 👍 

> The navigation editing feature in the sidebar does present a valid reason to switch the patterns to go via view mode.

I agree. If later we find it's mostly ending up as a blank sidebar (and that's deemed a problem) we can conditionalise this based on whether there is a Navigation in the pattern.

**Comment by getdave** [comment]
I'm going to wait on someone from @WordPress/gutenberg-design to approve / merge this one as @jameskoster has specifically tagged it.

**Comment by jasmussen** [comment]
This looks good to me. Before:

trunk

After:

this branch

**Comment by jameskoster** [comment]
For template parts – specifically headers – the details panel can be useful to see in this flow. 

The trade-off is that general template part and reusable block details panels are essentially empty:

Screenshot 2023-06-27 at 10 31 31

Which makes traversing them a little tedious. I suspect that's why the original design called for them to be skipped. 

It's not a blocker, but it would be nice to display some information in that panel. Author, usage, sync status, category, etc.

**Comment by aaronrobertshaw** [comment]
> It's not a blocker, but it would be nice to display some information in that panel. Author, usage, sync status, category, etc.

There are some minor updates in the pipeline on that front: https://github.com/WordPress/gutenberg/pull/51954

**Review by jameskoster — APPROVED** [review]
Nice. Let's give this a go then.

**Comment by ramonjd** [comment]
I just cherry-picked this PR to the release/16.1 branch to get it included in the next release: 8256ef9bfa

**Comment by tellthemachines** [comment]
I just cherry-picked this PR to the add/pre-beta1-bugfixes branch to get it included in the next release: 1bea96cdeb

**Pull request merged**
